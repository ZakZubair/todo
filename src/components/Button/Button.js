import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  onClick, children, className,
}) => (
  <button className={className} onClick={onClick} type="button">
    {children}
  </button>
);

Button.defaultProps = {
  children: null,
  className: 'button__add',
};

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};
export default Button;
