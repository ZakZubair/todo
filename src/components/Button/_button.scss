.button {
  &__add {
    font-family: $secondary-font;
    background: $turquoise;
    color: $emerald;
    font-size: rem(32);
    font-weight: 700;
    cursor: pointer;
    @include border-radius(100%);
    width: rem(60);
    height: rem(60);
    border: 2px solid $white;
    outline: 0;
    position: absolute;
    left: calc(50% - 30px);
    margin-top: -48px;
    transition: all 0.15s;

    &:hover {
      background: $emerald;
      color: $white;
    }

    &:disabled {
      background-color: $french-gray;
      color: $bright-gray;
      pointer-events: none;
    }
  }

  &__edit {
    color: $bright-gray;
    opacity: 0;
    background: transparent;
    border: none;
    padding: 0 rem(16);
    outline: 0;
    transition: all 0.15s;
    cursor: pointer;
    &:hover {
      text-decoration: underline;
    }
  }

  &__remove {
    font-size: rem(16);
    font-family: $secondary-font;
    color: $emerald;
    font-weight: 700;
    opacity: 0;
    width: rem(28);
    height: rem(28);
    border: rem(2) solid $athens-gray;
    background-color: $white;
    transition: all 0.15s;
    outline: 0;
    cursor: pointer;
    @include border-radius(100%);

    &:hover {
      background-color: $emerald;
      color: $white;
    }
  }
  // VCR button common elements
  &__vcr {
    width: rem(40);
    height: rem(40);
    border: 2px solid $french-gray;
    outline: 0;
    transition: all 0.15s;
    cursor: pointer;
    &:hover {
      opacity: 0.8;
    }
  }

  &__record {
    @extend .button__vcr; // Extend the common elements
    background: $alizarin-crimson;
    @include border-radius(100%);
  }

  &__stop {
    @extend .button__vcr; // Extend the common elements
    background-color: $bright-gray;
    @include border-radius(8px);
  }

  &__clear {
    @extend .button__vcr; // Extend the common elements
    border: 0;
    img {
      width: rem(40);
      height: 100%;
    }
  }

  &__play {
    @extend .button__vcr; // Extend the common elements
    background: transparent;
    box-sizing: border-box;
    border-width: rem(20) 0 rem(20) rem(36);
    border-color: transparent transparent transparent $japanese-laurel;
    &__playing {
      border-color: transparent transparent transparent $french-gray;
      pointer-events: none;
    }
  }
}
