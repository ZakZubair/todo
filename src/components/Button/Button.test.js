import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('test Button component', () => {
  it('test click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Button onClick={mockCallBack} />));
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('render correctly button component', () => {
    const mockCallBack = jest.fn();

    const ButtonComponent = renderer.create(<Button onClick={mockCallBack} />).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
  });
});
