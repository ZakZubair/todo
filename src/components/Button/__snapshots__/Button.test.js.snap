// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component render correctly button component 1`] = `
<button
  className="button__add"
  onClick={[MockFunction]}
  type="button"
/>
`;
